frontend/deb/pkg.go

```diff
@@ -0,0 +1,132 @@
+			Run(
+				dalec.ShArgs("set -e; git init .; git add .; git commit -m 'Initial commit'; \"${DEBIAN_DIR}/dalec/patch.sh\"; git add .; git commit -m 'With patch'; git diff HEAD~1 >> /work/out/dalec-changes.patch; echo 'dalec-changes.patch' > /work/out/series"),
+				llb.Dir("/work/sources"),
+				mountSources(sources, "/work/sources", nil),
+				// DEBIAN_DIR is used by the patch script to find the debian directory where we actually have the patches
+				llb.AddEnv("DEBIAN_DIR", "/work/debian"),
+				llb.AddMount("/work/debian", dr, llb.SourcePath("debian"), llb.Readonly),
+				dalec.WithConstraints(opts...),
+			).AddMount("/work/out", patchesMountInput)
+	}
+
+	return patches
+}
+
+func SourcePackage(sOpt dalec.SourceOpts, worker llb.State, spec *dalec.Spec, targetKey string, opts ...llb.ConstraintsOpt) (llb.State, error) {
+	dr, err := Debroot(sOpt, spec, llb.Scratch(), targetKey, "")
+	if err != nil {
+		return llb.Scratch(), err
+	}
+
+	sources, err := dalec.Sources(spec, sOpt)
+	if err != nil {
+		return llb.Scratch(), err
+	}
+
+	for _, patches := range spec.Patches {
+		for _, patch := range patches {
+			// patches do not need to be included in the sources
+			delete(sources, patch.Source)
+		}
+	}
+
+	patches := createPatches(spec, sources, worker, dr, opts...)
+	debSources := debSources(worker, sources)
+
+	work := worker.Run(
+		dalec.ShArgs("set -e; ls -lh; dpkg-buildpackage -S -us -uc; mkdir -p /tmp/out; ls -lh; cp -r /work/"+spec.Name+"_"+spec.Version+"* /tmp/out; ls -lh /tmp/out"),
+		llb.Dir("/work/pkg"),
+		llb.AddMount("/work/pkg/debian", dr, llb.SourcePath("debian")), // This cannot be readonly because the debian directory gets modified by dpkg-buildpackage
+		llb.AddMount("/work/pkg/debian/patches", patches, llb.Readonly),
+		llb.Network(pb.NetMode_NONE),
+		mountSources(sources, "/work/pkg", sanitizeSourceKey),
+		dalec.RunOptFunc(func(ei *llb.ExecInfo) {
+			for key, src := range debSources {
+				tarName := fmt.Sprintf("%s_%s.orig-%s.tar.gz", spec.Name, spec.Version, sanitizeSourceKey(key))
+				llb.AddMount(filepath.Join("/work", tarName), src, llb.SourcePath("src.tar.gz"), llb.Readonly).SetRunOption(ei)
+			}
+		}),
+		dalec.WithConstraints(opts...),
+	)
+
+	return work.AddMount("/tmp/out", llb.Scratch()), nil
+}
+
+func BuildDeb(worker llb.State, spec *dalec.Spec, sOpt dalec.SourceOpts, targetKey string, opts ...llb.ConstraintsOpt) (llb.State, error) {
+	srcPkg, err := SourcePackage(sOpt, worker, spec, targetKey)
+	if err != nil {
+		return llb.Scratch(), errors.Wrap(err, "error creating debian source package")
+	}
+
+	dirName := filepath.Join("/work", spec.Name+"_"+spec.Version+"-"+spec.Revision)
+	st := worker.
+		Run(
+			dalec.ShArgs("set -e; ls -lh; dpkg-source -x ./*.dsc; ls -lh; cd "+spec.Name+"-"+spec.Version+"; ls -lh *; dpkg-buildpackage -b -uc -us; mkdir -p /tmp/out; cp ../*.deb /tmp/out; ls -lh /tmp/out"),
```

Comment:
is the `ls -lh` for debugging?

Reply:
Yeah, when stuff goes south its nice to see the content there.